src/core/threadasm.S

```diff
@@ -406,21 +406,32 @@ fiber_switchContext:
 
 .text
 .align  2
-.global fiber_switchContext
+#ifdef __APPLE__
+.global _fiber_switchContext
+_fiber_switchContext:
+#else
 #if defined(__ARM_PCS_VFP) || (defined(__ARM_PCS) && !defined(__SOFTFP__)) // ARM_HardFloat  || ARM_SoftFP
```

Comment:
This should probably be exactly the same condition as in line 434 according to the comment. So extracting it as its own `#define` should pay off.

Reply:
Do you have a suggestion for the name of this `#define`?

Reply:
I *think* the condition is supposed to reflect whether FP registers need to be saved & restored when switching fibers. I *think* that's the case when hardware units are enabled (`version (D_HardFloat)`), even if FP registers aren't used to pass/return values when calling a function ('softfp' - use hardware units like hard-float, but be ABI-compatible to soft-float).

So if that's correct, something like `ARM_NonSoftFloat` might be workable.

Reply:
Fixed.